homeassistant/components/backup/http.py

```diff
@@ -70,6 +98,52 @@ async def get(
             await response.write(chunk)
         return response
 
+    async def _send_backup_with_password(
+        self,
+        hass: HomeAssistant,
+        request: Request,
+        headers: dict[istr, str],
+        backup_id: str,
+        agent_id: str,
+        password: str,
+        agent: BackupAgent,
+        manager: BackupManager,
+    ) -> StreamResponse | FileResponse | Response:
+        reader: IO[bytes]
+        if agent_id in manager.local_backup_agents:
+            local_agent = manager.local_backup_agents[agent_id]
+            path = local_agent.get_backup_path(backup_id)
+            try:
+                reader = await hass.async_add_executor_job(open, path.as_posix(), "rb")
+            except FileNotFoundError:
+                return Response(status=HTTPStatus.NOT_FOUND)
+        else:
+            stream = await agent.async_download_backup(backup_id)
+            reader = cast(IO[bytes], util.AsyncIteratorReader(hass, stream))
+
+        worker: threading.Thread | None = None
+        worker_done_event = asyncio.Event()
+
+        def on_done() -> None:
+            """Call by the worker thread when it's done."""
+            hass.loop.call_soon_threadsafe(worker_done_event.set)
+
+        try:
+            stream = util.AsyncIteratorWriter(hass)
+            worker = threading.Thread(
+                target=util.decrypt_backup, args=[reader, stream, password, on_done]
+            )
+            worker.start()
```

Comment:
I think it's enough to start the `try` block here before `worker.start()`. Then we can simplify the typing for `worker` and remove that check below.

Reply:
I didn't want to risk not closing the file. But I think you're right, if something goes wrong before we start the worker it means we're out of memory or something equally severe, and that's not something we are able to handle anyway.